ykcompile/src/lib.rs

```diff
@@ -97,7 +97,8 @@ pub struct RegAndOffset {
 
 #[derive(Debug, Clone, PartialEq)]
 pub enum Location {
-    Register(u8),
+    /// Register number, and if it contains a pointer.
+    Register(u8, bool),
```

Comment:
I think the codegen logic would be a little easier to follow if instead this was an enum, e.g.:
```
enum RegTy {
    Pointer,
    Immediate
}
```

Reply:
You mean in place of the `bool`. I have no preference here. @vext01 Thoughts?

Reply:
At least converting it from a tuple variant to a struct variant, so fields could have names would help. But that'd muddy this PR quite a bit, so save for later?

Reply:
Personally, I think an enum for this purpose is a lot more descriptive and harder to misuse, but it's not a hill I'll die on.

Reply:
I agree, but since I don't think it'll be a small change, could we vacate the hill until a later date? :P

Reply:
Works for me